pylint/checkers/refactoring.py

```diff
@@ -525,10 +529,72 @@ def _check_consider_using_in(self, node):
 
         self.add_message('consider-using-in', node=node, args=(suggestion,))
 
-    @utils.check_messages('consider-merging-isinstance', 'consider-using-in')
+    def check_chained_comparison(self, node):
+        """Check if there is any chained comparison in the expression.
+        If it exists, give message to simplify it."""
+        if node.op != 'and':
+            return
+
+        total_compares = len(node.values)
+        if total_compares < 2:
+            return
+
+        # all node values must be compare
+        for i in range(0, total_compares):
+            if not isinstance(node.values[i], astroid.node_classes.Compare):
+                return
+
+        # collect all compares as tuple
+        Comparison = collections.namedtuple('Comparison', ['left_operand',
```

Comment:
I'm not entirely convinced that we need a namedtuple for this. Also the algorithm might might need some simplifications as right now it feels a bit convoluted, and I can't exactly follow what it tries to do.

I suggest that we should look only for Name nodes and Consts (you don't currently check the other type in the `else` clause from line 557, it can be something else than a Const, for instance a function call).

Also instead of indexing elements, I suggest saving them in variables with suggestive names. Another thing, there's no need to iterate via an index, you can just iterate over the `node.values`. If you need the index `i` for indexing into other structure, feel free to use `zip()` instead.

I also feel that the whole lower bound / upper bound could be simplified, but I'll follow back later on that as I want to see an initial refactoring before coming with a new algorithm attempt.

Reply:
@PCManticore I was trying to work on these comments, but getting below error using pytest.

__________ test_functional[wrong_import_position_exclude_dunder_main] __________

test_file = <test_functional.FunctionalTestFile object at 0x6fffcec9cf8>

    @pytest.mark.parametrize("test_file", TESTS, ids=TESTS_NAMES)
    def test_functional(test_file):
        LintTest = LintModuleOutputUpdate(test_file) if UPDATE else LintModuleTe                                                                                                                st(test_file)
        LintTest.setUp()
>       LintTest._runTest()

test/test_functional.py:377:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test/test_functional.py:306: in _runTest
    self._linter.check([self._test_file.module])
lint.py:783: in check
    self._do_check(files_or_modules)
lint.py:916: in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
lint.py:996: in check_astroid_module
    walker.walk(ast_node)
utils.py:1140: in walk
    self.walk(child)
utils.py:1142: in walk
    cb(astroid)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <checkers.variables.VariablesChecker3k object at 0x6fff848a208>
node = <Assign l.2 at 0x6fff847ba90>

    def _store_type_annotation_names(self, node):
>       type_annotation = node.type_annotation
E       AttributeError: 'Assign' object has no attribute 'type_annotation'

checkers/variables.py:1338: AttributeError
============= 8 failed, 12 passed, 301 deselected in 6.42 seconds ==============

Using tox for testing even small changes is too much of work. It would be helpful if you could check whether def _store_type_annotation_names(self, node) is broken or not.